### ISSUE TITLE
json schema and json-edit

### ISSUE THREAD

**Issue by timelyportfolio** [daattali/shinyforms#19]
Great to see progress on this.  I think it will be a very valuable.  I'll attempt to write out some of the thoughts that I had in my head for #1.  I wondered if we could leverage JSON schema for validation/meta and then build/use something like [json-editor](https://github.com/jdorn/json-editor) or [alpaca](https://github.com/gitana/alpaca) to quickly create a usable form.  I did some tests a while ago auto-building a JSON schema from a R `list`, so you could have a 

sample input `list` -> create a JSON schema -> could be supplemented or customized by advanced users -> create a form that could be used with or without Shiny

I might be way off, but since you seemed very open to input, I thought I would share my very preliminary thoughts.

I had a list of some applicable JavaScript in https://github.com/timelyportfolio/buildingwidgets/issues/35.

**Comment by daattali** [comment]
Yup, that's exactly what I had in mind too (https://github.com/daattali/shinyforms/issues/11). Thanks for your input , I'll take a look at that link later.  I have to ditch all work on this package this month and focus a bit :)  I'll ping you when I need more help/input here

**Comment by timelyportfolio** [comment]
I should add this https://github.com/mozilla-services/react-jsonschema-form also.

**Comment by daattali** [comment]
Ooh pretty. That's the kind of thing I want to do in a gadget/addin that will help you create the form data

**Comment by jaeddy** [comment]
Hey @daattali - any updates on this item since 2016? Do you have a rough sense of how much effort would be required to incorporate functionality similar to `react-jsonschema-form`? I'd be happy to jump on a call sometime to discuss some possible funding for the work.

**Comment by daattali** [comment]
@jaeddy I haven't looked at this code in years and it's completely out of my radar, so I can't comment on how much work is needed. It would be great to return to this package at some point -- if you'd like to help make that happen feel free to email me more details or set up a chat using https://calendly.com/attalitech/meeting

**Comment by katrinabrock** [comment]
Did this conversation go anywhere last year?

I'm not familiar with shinyforms, but I'm weighing the pros and cons of building something that makes shiny inputs out of json schema vs just building a non-shiny react app and using existing packages. (Just building a react app would obviously be easier, but most of my team is R only so it would not be as maintainable by others.)

Would you be willing to provide guidance on design/review a PR for something like this if I decide to go the route?

If I hack something, I can also just post it even if it's not PR worthy. I don't expect I will have time/inclination/ability to implement every detail of json schema spec, but maybe the major parts..like data types.

**Comment by daattali** [comment]
> Did this conversation go anywhere last year?

It did not -- this project is still waiting for funding.

**Comment by katrinabrock** [comment]
I went ahead and build out creating a shiny input from a json schema as as separate package using reactR and react-jsonschema-form: https://github.com/katrinabrock/shinyjsonschemaform

Not sure if anyone following this thread is still interested, but it exists now. :-)

**Comment by timelyportfolio** [comment]
@katrinabrock very nicely done.  thank you.
